packages/react-native-usb/src/index.ts

```diff
@@ -0,0 +1,144 @@
+
+const releaseInterface = (deviceName: string, interfaceNumber: number) =>
+    ReactNativeUsbModule.releaseInterface(deviceName, interfaceNumber);
+
+const selectConfiguration = (deviceName: string, configurationValue: number) =>
+    ReactNativeUsbModule.selectConfiguration(deviceName, configurationValue);
+
+const transferIn = async (deviceName: string, endpointNumber: number, length: number) => {
+    const perf = performance.now();
+    const data = await ReactNativeUsbModule.transferIn(deviceName, endpointNumber, length)
+        .catch((error: any) => {
+            debugLog('JS: USB read error: ', error);
+            throw error;
+        })
+        // eslint-disable-next-line arrow-body-style
+        .then((result: number[]) => {
+            // debugLog('JS: Native USB read result:', JSON.stringify(result));
+
+            return {
+                data: new Uint8Array(result),
+                status: 'ok',
+            };
+        });
+    debugLog('JS: USB read time', performance.now() - perf);
+
+    return data;
+};
+
+const transferOut = async (
+    deviceName: string,
+    endpointNumber: number,
+    data: Uint8Array | BufferSource,
+) => {
+    try {
+        const perf = performance.now();
+        await ReactNativeUsbModule.transferOut(deviceName, endpointNumber, data.toString());
+        debugLog('JS: USB write time', performance.now() - perf);
+        return { status: 'ok' };
+    } catch (error) {
+        debugLog('JS: USB write error', error);
+        throw error;
+    }
+};
+
+// eslint-disable-next-line require-await
+const createNoop = (methodName: string) => async () => {
+    debugLog(`Calling ${methodName} which is not implemented.`);
+};
+
+const createWebUSBDevice = (device: NativeDevice): WebUSBDevice => ({
+    ...device,
+    open: () => open(device.deviceName),
+    close: () => close(device.deviceName),
+    forget: createNoop('forget'),
+    selectConfiguration: (configurationValue: number) =>
+        selectConfiguration(device.deviceName, configurationValue),
+    claimInterface: (interfaceNumber: number) => claimInterface(device.deviceName, interfaceNumber),
+    releaseInterface: (interfaceNumber: number) =>
+        releaseInterface(device.deviceName, interfaceNumber),
+    selectAlternateInterface: createNoop('selectAlternateInterface'),
+    controlTransferIn: createNoop('controlTransferIn'),
+    controlTransferOut: createNoop('controlTransferOut'),
+    clearHalt: createNoop('clearHalt'),
+    transferIn: (endpointNumber: number, length: number) =>
+        transferIn(device.deviceName, endpointNumber, length),
+    transferOut: (endpointNumber: number, data: BufferSource) =>
+        transferOut(device.deviceName, endpointNumber, data),
+    isochronousTransferIn: createNoop('isochronousTransferIn'),
+    isochronousTransferOut: createNoop('isochronousTransferOut'),
+    reset: createNoop('reset'),
+
+    // TODO: Implement these properties
```

Comment:
nit: Would be great to have an issue link for this as this is a big PR and will get lost IMO

Reply:
Good idea, but I probably wouldn't do that for this because it's super low priority because we don't using it anywhere anyway.

Reply:
I would consider removing the comment then? There is a lot of comments and todos so would be worth cleaning up a little bit if these things are low prio, let's just clean it up and push it as it is

Reply:
I added more extensive messages that explains why they are there and why it's low prio